db/app.js

```diff
@@ -1,11 +1,35 @@
-const{getTopics, getEndpoints} = require('./controllers/news.controllers')
+const{getTopics, getEndpoints, getArticlesById} = require('./controllers/news.controllers')
 const express = require('express')
 const app = express()
 
 app.get('/api/topics', getTopics)
 
 app.get('/api/', getEndpoints)
 
+app.get('/api/articles/:article_id', getArticlesById)
+
+
+
+app.use((error, request, response, next) => {
+    if(error.status && error.msg) {
+        response.status(error.status).send({msg: `${error.msg}`})
+} else next(error);
+})
+
+app.use((error, request, response, next) => {
+    console.log(error.code)
+    if(error.code === '22P02') {
+        response.status(400).send({msg: 'invalid input'})
+    } else next(error)
+    
+})
+
+app.use((error, request, response, next) => {
+    console.log(error);
+    response.status(500).send({msg: 'Internal Server Error'})
+})
+
```

Comment:
You may want to extract this error handlers into their own file at some point to keep things tidy :)

Reply:
Yup, was totally going to do that going forward as they were starting to get more complex, will do on the next ticket